packages/node/src/__tests__/integration.test.ts

```diff
@@ -39,16 +39,19 @@ describe('Error handling', () => {
     ).toThrowError(/writeKey/i)
   })
 
-  test('property validation errors are surfaced as thrown errors', async () => {
+  it('property validation errors are surfaced as thrown errors', async () => {
     const analytics = new AnalyticsNode({
       writeKey,
     })
     expect(() => analytics.track({} as any)).toThrowError(/event/i)
   })
 
-  test('http delivery errors are accessed through the emitter', (done) => {
+  it.skip('http delivery errors are accessed through the emitter', (done) => {
```

Comment:
I skipped this for now - it looked like any error in the previous plugin would end up being treated as an `http delivery error`, so I wanted to confirm if the intent was for this just to apply to errors thrown by `fetch` and not non-200 status codes. @silesky

Reply:
`http delivery` represents any non ok response along with any thrown   errors (that can usually contain a response)

Initially, almost all errors will be http delivery, but we may have things other error types like validation in the future, so it's a good future-proof pattern to have, and it also lets you add type-safe specific metadata based on the code (which is the discriminant key).